src/Models/NonhydrostaticModels/nonhydrostatic_model.jl

```diff
@@ -43,7 +43,7 @@ mutable struct IncompressibleModel{TS, E, A<:AbstractArchitecture, G, T, B, R, S
 end
 
 """
-    IncompressibleModel(;
+    NonhydrostaticModel(;
```

Comment:
Would be good to modify the description of this user-facing function as well. Right now it only says "Construct an incompressible `Oceananigans.jl` model on `grid`."

Maybe something like: Constructs an incompressible non-hydrostatic model. All boundaries are rigid and the fluid is assumed to satisfy the Boussinesq approximation whenever a buoyancy model is used.

(I couldn't suggest this as a possible commit for some reason...)

Reply:
Good idea to make it more informative! It's silly to call it an "Oceananigans.jl model"...

Here's a slight modification to @tomchor's suggestion:

"Construct a model for a non-hydrostatic, incompressible fluid, using the Boussinesq approximation when `buoyancy != nothing`. By default, all `Bounded` directions are rigid and impenetrable."

I think we want to be careful not to discourage exploration. In principle, an intrepid user can actually implement a model for a linear free surface (with in flows and outflows) at the top. I do think it's useful to mention default boundary conditions though.

Regarding the Boussinesq approximation, I don't think the use of Boussinesq is an assumption per se. Any assumptions regarding the Boussinesq approximation occur outside the code (eg an assumption that the numerical model provides a representation of reality --- certainly one should be careful about _that_)!

"Construct" might be better than "Constructs" just because it's a [julia recommendation](https://docs.julialang.org/en/v1/manual/documentation/) (but not a passionate issue for me).